src/main/java/uk/gov/companieshouse/controller/DissolutionController.java

```diff
@@ -136,4 +137,33 @@ public DissolutionPatchResponse patchDissolutionApplication(
             throw new NotFoundException();
         }
     }
+
+    @PatchMapping("/directors/{director-id}")
+    @ResponseStatus(HttpStatus.OK)
+    public DissolutionPatchResponse patchDissolutionDirector(
+            @RequestHeader("ERIC-Authorised-User") String authorisedUser,
+            @PathVariable("company-number") final String companyNumber,
+            @PathVariable("director-id") final String directorId,
+            @Valid @RequestBody final DissolutionPatchDirectorRequest body
+    ) {
+
+        if (!dissolutionService.doesDissolutionRequestExistForCompanyByCompanyNumber(companyNumber)
+                || !dissolutionService.doesDirectorExist(companyNumber, directorId)) {
```

Comment:
are both of these methods individually querying mongo? If so, consider creating a function that can call the db once and check both conditions.